### PR TITLE
Docker update

### LINKED ISSUE #71: Add canlang to the Docker image
Add canlang to the Docker image, a specific release of it, so that we have a stable version of the data sets. When we do this, we should delete https://github.com/UBC-DSCI/introduction-to-datascience/blob/dev/data/create_can_lang_variants.R

### PR DESCRIPTION
**Pull request by ttimbers**
Update to Dockerfile and actions that control its building.
- Only build Docker image when we push to master (makes cleaner image tags)
- Added canlang R package and LaTeX packages
- Added `docker-compose.yml` so that we can launch the RStudio container via `docker-compose up -d` to launch the rstudio web app, and `docker-compose run --rm rstudio bash` to run the container with bash as the entry point. This is less to type, and correctly specifies the ports, volume path and env vars.

Closes #71 and  #84.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +34,12 @@ This is the source for the Introduction to Data Science textbook.
 
 > ### Usage without RStudio
 > You can use this docker container to edit the files without RStudio using either vim or emacs. This can be done via:
+>
+> ```
+> docker-compose run --rm rstudio bash
```
Comment (trevorcampbell) [inline README.md:39]: this shouldn't be in the "using without Rstudio" section, right?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+services:
+  rstudio:
+    image: "ubcdsci/intro-to-ds:v0.6.0-47d5cbf"
```
Comment (trevorcampbell) [inline docker-compose.yml:3]: is this docker-compose.yml file auto-generated?

- if so: do we want it to be committed to the repo?
- if not: do we want that `v0.6.0-47blahblah` to be hard-coded?

**Review by trevorcampbell — COMMENTED** [review]
looks good to me -- i just have a few confusions that are almost certainly because i'm a GH workflows novice.

File: README.md (hunk 1 above)
Comment (ttimbers) [reply README.md:39]: It should be, rstudio is just the nickname for the container. But now this makes me think this is not the best nickname... I'll change this to  `book-env`.

File: docker-compose.yml (hunk 2 above)
Comment (ttimbers) [reply docker-compose.yml:3]: b5c491e and f16fd62 should fix this

File: README.md (hunk 1 above)
Comment (ttimbers) [reply README.md]: 90bf19c and 4184d39 address this.

**Comment by ttimbers** [comment]
OK - still need to fix this! I am currently just printing the regex and not updating the file 🤦‍♀️ !

**Comment by ttimbers** [comment]
fixed here: 72c7358

**Pull request merged**